### PR TITLE
sg: Add description field to sg commands

### PR DESCRIPTION
**Pull request by marekweb**
Add a `description` field to `sg` commands specified in `sg.config.yaml`

The description is displayed in the list of commands when running `sg run`

## Test plan

- Run `sg run`
- See that the command list now shows a description, when provided in `sg.config.yaml`

### CODE REVIEW

File: dev/sg/sg_run.go [hunk 1]
```diff
@@ -93,7 +93,10 @@ func constructRunCmdLongHelp() string {
 	fmt.Fprintf(&out, "Available commands in `%s`:\n", configFile)
 
 	var names []string
-	for name := range config.Commands {
+	for name, command := range config.Commands {
+		if command.Description != "" {
+			name = fmt.Sprintf("%s (%s)", name, command.Description)
```
Comment (bobheadxi) [inline dev/sg/sg_run.go:98]:
```suggestion
			name = fmt.Sprintf("%s: %s", name, command.Description)
```

File: dev/sg/internal/run/command.go [hunk 2]
```diff
@@ -30,7 +30,7 @@ type Command struct {
 	Preamble string `yaml:"preamble"`
 
 	ExternalSecrets map[string]secrets.ExternalSecret `yaml:"external_secrets"`
-
+	Description     string                            `yaml:"description"`
```
Comment (bobheadxi) [inline dev/sg/internal/run/command.go:33]:
```suggestion
	Description     string                            `yaml:"description"`

```

we might want to preserver the new line here

File: sg.config.yaml [hunk 3]
```diff
@@ -116,6 +116,7 @@ env:
 
 commands:
   server:
+    description: Run a server image
```
Comment (bobheadxi) [inline sg.config.yaml:119]:
```suggestion
    description: Run an all-in-one sourcegraph/server image
```

File: doc/dev/background-information/sg/reference.md [hunk 4]
```diff
@@ -100,21 +100,21 @@ Available commands in `sg.config.yaml`:
...
 * oss-worker
-* otel-collector
+* otel-collector: OpenTelemetry collector
 * postgres_exporter
 * prometheus
 * redis-postgres
 * repo-updater
 * searcher
-* server
+* server: Run an all-in-one sourcegraph/server image
```
Comment (jhchabran) [inline doc/dev/background-information/sg/reference.md:111]: This is a great idea! While it's kind of common knowledge that the server refers to the all in one instance, the description adds value in there.

File: doc/dev/background-information/sg/reference.md [hunk 5]
```diff
@@ -100,21 +100,21 @@ Available commands in `sg.config.yaml`:
...
 * oss-repo-updater
 * oss-symbols
-* oss-web
+* oss-web: Open source version of the web app
 * oss-worker
-* otel-collector
+* otel-collector: OpenTelemetry collector
 * postgres_exporter
 * prometheus
 * redis-postgres
```
Comment (jhchabran) [inline doc/dev/background-information/sg/reference.md:108]: _Dockerized version of redis and postgres_

File: doc/dev/background-information/sg/reference.md [hunk 6]
```diff
@@ -100,21 +100,21 @@ Available commands in `sg.config.yaml`:
...
 * repo-updater
 * searcher
-* server
+* server: Run an all-in-one sourcegraph/server image
 * storybook
 * symbols
 * syntax-highlighter
-* web
 * web-standalone-http
```
Comment (jhchabran) [inline doc/dev/background-information/sg/reference.md:115]: This and the following one would benefit from having a description too!

**Review by jhchabran — APPROVED** [review]
Nice idea, I think this will help discoverability!

File: doc/dev/background-information/sg/reference.md (hunk 6 above)
Comment (bobheadxi) [reply doc/dev/background-information/sg/reference.md]: Yeah all the web stuff could do with some descriptions, cc @sourcegraph/frontend-platform-devs - but I don't think it needs to be in this PR :)

**Pull request merged**